### PR TITLE
ticdc: add kafka oauth configurations

### PR DESCRIPTION
**Pull request by hi-rustin**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

add kafka oauth configurations.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tiflow/pull/8938

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: ticdc/ticdc-changefeed-config.md [hunk 1]
```diff
@@ -198,4 +198,21 @@ use-file-backend = false
 integrity-check-level = "none"
 # 当单行数据的 Checksum 校验失败时，Changefeed 打印错误行数据相关日志的级别。默认值为 "warn"，可选值为 "warn" 和 "error"。
 corruption-handle-level = "warn"
+
+# 以下参数仅在下游为 Kafka 时生效。
```
Comment (Oreoxmt) [inline ticdc/ticdc-changefeed-config.md:202]:
```suggestion
# 以下参数仅在下游为 Kafka 时生效。从 v6.5.3 开始支持。
```

File: ticdc/ticdc-changefeed-config.md [hunk 2]
```diff
@@ -198,4 +198,21 @@ use-file-backend = false
...
+[sink.kafka-config]
+# Kafka SASL 认证机制。默认值为空。当该参数为空时，表示不使用 SASL 认证。
+sasl-mechanism = "OAUTHBEARER"
+# Kafka SASL OAUTHBEARER 认证机制中的 client-id。默认值为空。在使用该认证机制时，该参数必填。
+sasl-oauth-client-id = "producer-kafka"
+# Kafka SASL OAUTHBEARER 认证机制中的 client-secret。默认值为空。需要 base64 编码。在使用该认证机制时，该参数必填。
```
Comment (Oreoxmt) [inline ticdc/ticdc-changefeed-config.md:208]:
```suggestion
# Kafka SASL OAUTHBEARER 认证机制中的 client-secret。默认值为空。需要 Base64 编码。在使用该认证机制时，该参数必填。
```

File: ticdc/ticdc-changefeed-config.md [hunk 3]
```diff
@@ -198,4 +198,21 @@ use-file-backend = false
 integrity-check-level = "none"
 # 当单行数据的 Checksum 校验失败时，Changefeed 打印错误行数据相关日志的级别。默认值为 "warn"，可选值为 "warn" 和 "error"。
 corruption-handle-level = "warn"
+
+# 以下参数仅在下游为 Kafka 时生效。
+[sink.kafka-config]
+# Kafka SASL 认证机制。默认值为空。当该参数为空时，表示不使用 SASL 认证。
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:204]:
```suggestion
# Kafka SASL 认证机制。该参数默认值为空，表示不使用 SASL 认证。
```

File: ticdc/ticdc-changefeed-config.md [hunk 4]
```diff
@@ -198,4 +198,21 @@ use-file-backend = false
 integrity-check-level = "none"
 # 当单行数据的 Checksum 校验失败时，Changefeed 打印错误行数据相关日志的级别。默认值为 "warn"，可选值为 "warn" 和 "error"。
 corruption-handle-level = "warn"
+
+# 以下参数仅在下游为 Kafka 时生效。从 v6.5.3 开始支持。
```
Comment (hi-rustin) [inline ticdc/ticdc-changefeed-config.md:202]:
```suggestion
# 以下参数仅在下游为 Kafka 时生效。
```

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**